### ISSUE TITLE
Some unexpected purchases id

### ISSUE THREAD

**Issue by tsimox** [onepf/OpenIAB#484]
Hi. I'm using OpenIAB for Google Play, skipping verificatoin because of checking it on server.
So I've noticed some unexpected things.

Receiving a purchase, then through:
_String receiptData = purchase.getOriginalJson();_
I am sending this string to server.

But in server log I see some junk events with wrong transactoin IDs like
_{"orderId":"6692965933411890189",...}_
It is even not suits Google In-app format.

Bad thing that I am missing some REAL purchases.
Cant figure it out. Already broke my mind trying to understand.

**Comment by tsimox** [comment]
Sometimes I am catching other application's events, with their ids in Google Play format like
_4242245428367783384.5130579051861852_

**Comment by RomanZhilich** [comment]
Maybe this are unfair purchases made with Freedom or something?
I have no idea how this could happen otherwise, `origialJson` comes form Google.
>I am catching other application's events

I'm not sure what you mean by that.

**Comment by tsimox** [comment]
Every day I reveive 60 junk events with id like 6692965933411890189, and couple of events like real 4242245428367783384.5130579051861852.

As you can know first part of id - application id. But it isn't mine.

**Comment by RomanZhilich** [comment]
Might be a Google's bug. However this is very strange, I'm pretty sure you can't get purchases that doesn't belong to your app.
Maybe this ids comes from another Appstore?

**Comment by tsimox** [comment]
No, they are google style so I suppose they belong to Google Play.
Junk events - not a real problem. Real problem that I am not tracking some _my_ purchases (I see them in Wallet).
Main problem - that "some". When I'm testing, or doing real transactions it works. Like 90% of customers transactions.
But there is a 10% of untracked purchases, and I dont know, how to handle them.

**Comment by tsimox** [comment]
I've noticed that I am not tracking purchases from USA. Are there some ideas? Maybe they have some special tricks connected with specified cards. It blows my mind. 
Common flow is not working for me.
I think it because of VERIFICATION_SKIP, it is not usual thing, not tested enough.

**Comment by RomanZhilich** [comment]
> VERIFICATION_SKIP

Is that what you're using? I'm pretty sure it should be fine. Actually there's not much that can go wrong with "not doing stuff", in this case not checking purchases.

> Maybe they have some special tricks connected with specified cards.

This part should be handled by Google, it hardly has anything to do with library.

> there is a 10% of untracked purchases

Maybe the problem is with "tracking" purchases? How do you report purchase event to your analytics?

**Comment by tsimox** [comment]
Thanks for your interest, Roman, and sorry for late response.

The tracking algorithm is the following, contains two branches:
1. Billing Activity receiving purchase callback from OpenIAB and send data to our server. As server answer is OK (received) purchase consumed.
2. Billing Activity retrieving asynchronously  unconsumed purchases (if previous step wasnt finished properly) and send it to server. And like in the first approach as soon as server answer is OK - purchase will be consumed.

I was rewrite and refactor this code a week ago, now it seems working (no missed purchases).
But there are still a lot of junk stuff (wrong purchases with wrong ids).

**Comment by RomanZhilich** [comment]
>  now it seems working (no missed purchases)

Glad it worked out.

> But there are still a lot of junk stuff

Unfortunately I have no idea where they come from. If you'll be able to reproduce this "junk" purchase, please let me know.

**Comment by tsimox** [comment]
I'm just wondering do you know anyone, who perform server-side checking like me? Do they receive the same? Or it is my-application-specific.
